src/main/groovy/org/shipkit/internal/gradle/git/GitPullTask.java

```diff
@@ -0,0 +1,50 @@
+package org.shipkit.internal.gradle.git;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+
+public class GitPullTask extends DefaultTask{
+
+    @Input private String url;
+    @Input private String rev;
+    @Input private boolean dryRun;
+    private String secretValue;
+
+    @TaskAction
+    public void gitPull(){
+        new GitPull().gitPull(this);
```

Comment:
Very nice, thank you!!! We push the internal implementation out of the public type.